### PR TITLE
Add more statistics

### PR DESCRIPTION
**Pull request by matt-frey**
This PR
* adds minimum and maximum buoyancy evaluation for fields and parcels
* adds enstrophy calculation on the grid
* removes some deprecated code and comments
* renames `psi` to `en` for consistency

### CODE REVIEW

File: src/3d/fields/field_diagnostics_netcdf.f90 [hunk 1]
```diff
@@ -144,6 +146,36 @@ subroutine create_netcdf_field_stats_file(basename, overwrite, l_restart)
                 dimids=(/t_dim_id/),                                        &
                 varid=keg_id)
 
+            call define_netcdf_dataset(                                     &
+                ncid=ncid,                                                  &
+                name='en',                                                  &
```
Comment (sjboeing) [inline src/3d/fields/field_diagnostics_netcdf.f90:151]: Note psi is used for enstrophy in the parcel diagnostics.

File: src/3d/epic3d.f90 [hunk 2]
```diff
@@ -141,7 +141,6 @@ subroutine run
 #endif
             double precision :: t = zero    ! current time
             integer          :: cor_iter    ! iterator for parcel correction
-            integer :: n_orig
```
Comment (sjboeing) [inline src/3d/epic3d.f90:144]: I wouldn't mind keeping these diagnostics in, but maybe the default diagnostics are sufficient.

**Review by sjboeing — COMMENTED** [review]
Thanks Matthias, I have added a few notes but am happy for these changes to be merged in.

File: src/3d/epic3d.f90 (hunk 2 above)
Comment (matt-frey) [reply src/3d/epic3d.f90:144]: I only used these for debugging purposes and forgot to remove them. I think we now have the diagnostics written to NetCDF with `n_parcel_merges` and `n_parcel_splits`.

File: src/3d/fields/field_diagnostics.f90 [hunk 3]
```diff
@@ -16,7 +16,10 @@ module field_diagnostics
                         min_npar,   &       ! min num parcels per cell
                         avg_npar,   &       ! average num parcels per cell
                         avg_nspar,  &       ! average num small parcels per cell
-                        keg                 ! kinetic energy calculated on the grid
+                        keg,        &       ! kinetic energy calculated on the grid
+                        eng,        &       ! enstrophy calculated on the grid
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/fields/field_diagnostics.f90:20]: Changed that now.

File: src/3d/epic3d.f90 (hunk 2 above)
Comment (sjboeing) [reply src/3d/epic3d.f90:144]: Yes, given they are in the diagnostics, it is fine to remove this here.

File: src/3d/fields/field_diagnostics.f90 (hunk 3 above)
[the remark this answers is not in the document]
Comment (sjboeing) [reply src/3d/fields/field_diagnostics.f90:20]: Sounds good (realise you will also need to change your plotting scripts).

[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/fields/field_diagnostics.f90:20]: Indeed, but this is a minor thing. I have a branch to update the scripts for the 3D paper anyway where I add new plotting scripts.

**Review by sjboeing — APPROVED** [review]

**Pull request merged**